File: Trabajo2/CodigoEjercicio2.java
import java.util.Scanner;

class Persona {
    private String nombre;
    private String apellido;
    private String sexo;
    private int edad;

    // Constructor
    public Persona() {
        nombre = "";
        apellido = "";
        sexo = "";
        edad = 0;
    }

    public void pedirDatos() {
        Scanner scanner = new Scanner(System.in);

        System.out.println("\n=== INGRESO DE DATOS ===");
        System.out.print("Nombre: ");
        nombre = scanner.nextLine();

        System.out.print("Apellido: ");
        apellido = scanner.nextLine();

        System.out.print("Sexo (M/F): ");
        sexo = scanner.nextLine();

        System.out.print("Edad: ");
        edad = scanner.nextInt();
        scanner.nextLine(); // Limpiar el buffer del scanner
    }

    public void mostrarDatos() {
        System.out.println("\n=== DATOS DE LA PERSONA ===");
        System.out.println("Nombre: " + nombre);
        System.out.println("Apellido: " + apellido);
        System.out.println("Sexo: " + sexo);
        System.out.println("Edad: " + edad + " años");
        System.out.println("============================");
    }
}

public class Main {
    public static void main(String[] args) {
        Persona[] personas = new Persona[2];
        Scanner scanner = new Scanner(System.in);

        System.out.println("SISTEMA DE REGISTRO DE PERSONAS");
        System.out.println("Ingrese los datos de 2 personas:");

        // Pedir y mostrar datos
        for(int i = 0; i < 2; i++) {
            System.out.println("\nPersona " + (i + 1) + ":");
            personas[i] = new Persona();
            personas[i].pedirDatos();
            personas[i].mostrarDatos();
        }

        System.out.println("\nRegistro completado exitosamente!");
        scanner.close();
    }
}

File: Trabajo2/CodigoEjercicio 2.cpp
#include <iostream>
#include <string>
using namespace std;

class Persona {
private:
    string nombre, apellido, sexo;
    unsigned int  edad;

public:
    Persona();  // Constructor
    void pedirDatos();
    void mostrarDatos();
};

// Implementación del constructor
Persona::Persona() {
    nombre = "";
    apellido = "";
    sexo = "";
    edad = 0;
}

// Implementación de pedirDatos
void Persona::pedirDatos() {
    cout << "\n=== INGRESO DE DATOS ===" << endl;
    cout << "Nombre: ";
    getline(cin >> ws, nombre);  // ws para limpiar buffer

    cout << "Apellido: ";
    getline(cin, apellido);

    cout << "Sexo (M/F): ";
    getline(cin, sexo);

    cout << "Edad: ";
    cin >> edad;

}

// Implementación de mostrarDatos
void Persona::mostrarDatos() {
    cout << "\n=== DATOS DE LA PERSONA ===" << endl;
    cout << "Nombre: " << nombre << endl;
    cout << "Apellido: " << apellido << endl;
    cout << "Sexo: " << sexo << endl;
    cout << "Edad: " << edad << " años" << endl;

    cout << "============================" << endl;
}

int main() {
    Persona personas[2];

    cout << "SISTEMA DE REGISTRO DE PERSONAS" << endl;
    cout << "Ingrese los datos de 2 personas:" << endl;

    // Pedir y mostrar datos
    for(int i = 0; i < 2; i++) {
        cout << "\nPersona " << (i + 1) << ":" << endl;
        personas[i].pedirDatos();
        personas[i].mostrarDatos();
    }

    cout << "\nRegistro completado exitosamente!" << endl;

    return 0;
}